woocommerce/woocommerce
# Added feature for payment gateways banner

### All Submissions:






### Changes proposed in this Pull Request:



This PR makes a set of changes in order to implement the feature for a payment gateways banner.

There were two possible approaches to implement this - fully in PHP and the other to embed a React tree on a new element that is introduced on the PHP side.

The decision was made to embed a React tree for ease of future maintenance, and also to make the new mount point an experimental Slotfill so that we can add future items to it.

Explat post: pbxNRc-1yD-p2

Closes #32414 .

###### Control:

![image](https://user-images.githubusercontent.com/27843274/164360930-dc67b527-6ae7-4a1f-9bc7-f1b773da7518.png)

###### Treatment: 

![image](https://user-images.githubusercontent.com/27843274/164361604-0bb67e2a-b1fc-44bb-9f69-bfd1591e5d46.png)

### How to test the changes in this Pull Request:

There are 5 scenarios to test:
- WCPay eligible and WCPay installed and setup completed and assigned to 'treatment' -> Should not see the banner
- WCPay eligible and WCPay not installed -> Should not see the banner
- WCPay eligible and WCPay installed but setup incomplete and assigned to 'treatment' -> Should see the banner
- WCPay eligible and WCPay installed but setup incomplete and assigned to 'control' -> Should not see the banner
- WCPay ineligible and/or WCPay not installed -> Should not see the banner

1. Set up a fresh install of Wordpress with WooCommerce, and depending on the scenario you are testing, install WooCommerce Payments plugin (either as part of onboarding or otherwise)
2. Install and activate [wc-test-helper](https://github.com/woocommerce/woocommerce-admin-test-helper/releases/tag/v0.7.2) plugin
3. Navigate to Tools -> WCA Test Helper
4. Click **Experiments**
5. Toggle the experiment assignment for 'woocommerce_payments_settings_banner_2022_04' to control or treatment depending on the scenario
6. Visit WooCommerce > Settings, Payments tab, and check if the banner is visible or not 

### Other information:




### FOR PR REVIEWER ONLY:
